### DISCUSSION TITLE
Use GitHub's Merge Queue (compatible with Renovate) to make merges easier for the maintainers

### DISCUSSION THREAD

**Discussion by HonkingGoose** [eslint/eslint#17582] [Ideas] [answered]
You may want to look into using GitHub's Merge Queue. I see you're not using it...

The main Renovate repository uses the Merge Queue, and it makes merging easier and faster. I don't need to update my branches when `main` changes under me after PR approval. The GitHub Merge Queue will auto-update branches, without making any more noise.

Read the [GitHub Docs, How merge queues work](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#how-merge-queues-work) to learn the specifics.

@JoshuaKGoldberg is working on the Renovate config for your repository:

- #17567

Renovate supports GitHub Merge Queue. Learn how to get [Renovate working with GitHub's Merge Queue in the Renovate docs](https://docs.renovatebot.com/key-concepts/automerge/#github-merge-queue).

You'll need to ask somebody else for help to get GitHub Actions configured correctly for the Merge Queue. 😉

**Comment by nzakas** [comment]
> The main Renovate repository uses the Merge Queue, and it makes merging easier and faster.

Thanks for the suggestion, but this doesn't really give us any details on which to base a decision. What exactly is easier and faster? What problems were you experiencing before that you no longer experience? 

In general, we don't see to have many problems merging PRs, so I'm not sure what benefits we'd get.

**Reply by HonkingGoose** [reply]
We require all PR branches to be current with `main` before they are merged. With the default GitHub automerge, here's what getting a PR merged looked like:

1. @HonkingGoose creates PR to update Renovate docs
1. Maintainer approves PR
1. Maintainer enables automerge
1. `main` branch gets a new commit
1. My PR branch is behind `main` now, so I select the "Update branch" button to get current with `main` again
1. Because the PR has a new commit (that happens _after_ approval) the PR is no longer approved
1. Request a new review from the maintainer 
1. Maintainer approves PR _again_
1. Maintainer enables automerge _again_
1. Pray `main` stays where it is now, so GitHub can automerge the PR

Sometimes getting the PR merged needed multiple "update branch and approve"-cycles. This was obviously wasting a lot of maintainer time and brainpower.

The Renovate team wanted a "keep rebasing this PR until it's merged" option, and that's what Merge Queue gave us.

To summarize GitHub's Merge Queue: "Put PR in Merge Queue and forget about it, GitHub will keep it up to date, and only merge PRs that pass all tests".

**Reply by nzakas** [reply, marked as answer]
Ah I see, thanks for explaining. We don't require PRs to be current with `main` before merging, so not sure how much value it'll have for us, but we can take a look.
